org.mwc.debrief.track_shift/src/org/mwc/debrief/track_shift/controls/ZoneChart.java

```diff
@@ -78,6 +78,54 @@ private CustomChartComposite(final Composite parent, final JFreeChart chart)
 
     private Stack<Rectangle> zoomHistory = new Stack<Rectangle>();
 
+    void fitToData()
+    {
+      final List<Rectangle> zoomRedoHistory = new ArrayList<>(zoomHistory);
+
+      AbstractOperation addOp = new AbstractOperation("Show all data")
+      {
+
+        @Override
+        public IStatus execute(IProgressMonitor monitor, IAdaptable info)
+            throws ExecutionException
+        {
+
+          zoomHistory.clear();
+
+          CustomChartComposite.super.restoreAutoBounds();
+          return Status.OK_STATUS;
+        }
+
+        @Override
+        public IStatus redo(IProgressMonitor monitor, IAdaptable info)
+            throws ExecutionException
+        {
+          zoomHistory.clear();
+
+          CustomChartComposite.super.restoreAutoBounds();
+          return Status.OK_STATUS;
+        }
+
+        @Override
+        public IStatus undo(IProgressMonitor monitor, IAdaptable info)
+            throws ExecutionException
+        {
+
+          CustomChartComposite.super.restoreAutoBounds();
+          zoomHistory.addAll(zoomRedoHistory);
+
+          for (Rectangle h : zoomHistory)
+          {
```

Comment:
Well issue seems on zoom I get rectangle relative to it current zoom state than from fit to window. so I need to work like a macro to correctly redo  multiple steps